versions/3.1.1.md

```diff
@@ -1655,26 +1655,77 @@ requestBody:
 
-```yaml
-requestBody:
-  content:
-    multipart/form-data:
-      schema:
-        properties:
-          # The property name 'file' will be used for all files.
-          file:
-            type: array
-            items: {}
-```
+##### Support for x-www-form-urlencoded Request Bodies
 
-As seen in the section on `multipart/form-data` below, the empty schema for `items` indicates a media type of `application/octet-stream`.
+See [Encoding the `x-www-form-urlencoded` Media Type](#encoding-the-x-www-form-urlencoded-media-type) for guidance and examples, both with and without the `encoding` attribute.
 
-##### Support for x-www-form-urlencoded Request Bodies
+##### Special Considerations for `multipart` Content
+
+See [Encoding `multipart` Media Types](#encoding-multipart-media-types) for further guidance and examples, both with and without the `encoding` attribute.
+
+#### <a name="encodingObject"></a>Encoding Object
+
+A single encoding definition applied to a single schema property.
+See [Appendix B](#dataTypeConversion) for a discussion of converting values of various types to string representations.
+
+Properties are correlated with `multipart` parts using the `name` parameter to `Content-Disposition: form-data`, and with `application/x-www-form-urlencoded` using the query string parameter names.
+In both cases, their order is implementation-defined.
+
+See [Appendix E](#percentEncodingAndFormMediaTypes) for a detailed examination of percent-encoding concerns for form media types.
+
+##### Fixed Fields
+
+###### Common Fixed Fields
+
+These fields MAY be used either with or without the RFC6570-style serialization fields defined in the next section below.
+
+Field Name | Type | Description
+---|:---:|---
+<a name="encodingContentType"></a>contentType | `string` | The `Content-Type` for encoding a specific property. The value is a comma-separated list, each element of which is either a specific media type (e.g. `image/png`) or a wildcard media type (e.g. `image/*`). Default value depends on the property type as shown in the table below.
+<a name="encodingHeaders"></a>headers | Map[`string`, [Header Object](#headerObject) \| [Reference Object](#referenceObject)] | A map allowing additional information to be provided as headers. `Content-Type` is described separately and SHALL be ignored in this section. This property SHALL be ignored if the request body media type is not a `multipart`.
+
+This object MAY be extended with [Specification Extensions](#specificationExtensions).
+
+The default values for `contentType` are as follows, where an _n/a_ in the `contentEncoding` column means that the presence or value of `contentEncoding` is irrelevant:
+
+Property `type` | Property `contentEncoding` | Default `contentType`
+--------------- | -------------------------- | ---------------------
+_absent_ | _n/a_ | `application/octet-stream`
```

Comment:
This line is not present in 3.0.4: why don't we need it there?

Reply:
Because `type` being absent has a specific meanint (see "Working With Binary Data") in draft 2020-12 / 3.1, which it does not in draft-04 / 3.0.  It is used for instances (such as binary data) that do not fit the JSON Schema data model.  That guidance was added to 2020-12 specifically to enable using the `content*` keywords with data that does not conform to the JSON data model at all.

Reply:
@ralfhandl honestly, I wanted to completely rearrange all of the data model things to put all of the rules in one place, but it would be a huge amount of work and would require a specification with a very different organization, so I decided against it.